integrate doesn't work with >2 variables

```jl
julia> x, y = set_variables("x y")
2-element Array{TaylorSeries.TaylorN{Float64},1}:
  1.0 x + 𝒪(‖x‖⁷)
  1.0 y + 𝒪(‖x‖⁷)

julia> integrate(x, 1)   # 2 variables seems to work
 0.5 x² + 𝒪(‖x‖⁷)

julia> x, y, z = set_variables("x y z")
3-element Array{TaylorSeries.TaylorN{Float64},1}:
  1.0 x + 𝒪(‖x‖⁷)
  1.0 y + 𝒪(‖x‖⁷)
  1.0 z + 𝒪(‖x‖⁷)

julia> integrate(z, 1)   # 3 variables doesn't
ERROR: AssertionError: order ≤ get_order() && lencoef ≤ num_coeffs
Stacktrace:
 [1] resize_coeffsHP!(::Array{Float64,1}, ::Int64) at /Users/dpsanders/.julia/v0.6/TaylorSeries/src/auxiliary.jl:37
 [2] Type at /Users/dpsanders/.julia/v0.6/TaylorSeries/src/constructors.jl:96 [inlined]
 [3] integrate(::TaylorSeries.HomogeneousPolynomial{Float64}, ::Int64) at /Users/dpsanders/.julia/v0.6/TaylorSeries/src/calculus.jl:316
 [4] integrate(::TaylorSeries.TaylorN{Float64}, ::Int64) at /Users/dpsanders/.julia/v0.6/TaylorSeries/src/calculus.jl:333

```

--- a/src/calculus.jl
+++ b/src/calculus.jl
@@ -298,9 +298,9 @@ function integrate(a::HomogeneousPolynomial, r::Int)
     a.order == order_max && return HomogeneousPolynomial(zero(T), 0)
 
     @inbounds posTb = pos_table[a.order+2]
-    @inbounds num_coeffs = size_table[a.order+2]
-    coeffs = zeros(T, size_table[num_coeffs])
     @inbounds num_coeffs = size_table[a.order+1]
+
+    coeffs = zeros(T, size_table[a.order+2])
 
     @inbounds for i = 1:num_coeffs
         iind = coeff_table[a.order+1][i]

--- a/test/manyvariables.jl
+++ b/test/manyvariables.jl
@@ -159,6 +159,9 @@ end
     @test integrate(xT^17, 1, 2.0) == 2.0
     @test_throws AssertionError integrate(xT, 1, xT)
 
+
+
+
     @test derivative(2xT*yT^2,1) == 2yT^2
     @test xT*xT^3 == xT^4
     txy = 1.0 + xT*yT - 0.5*xT^2*yT + (1/3)*xT^3*yT + 0.5*xT^2*yT^2
@@ -544,4 +547,17 @@ end
     @test yT[0] == xT[0]*(pi/180)
     TaylorSeries.rad2deg!(yT, xT, 0)
     @test yT[0] == xT[0]*(180/pi)
+end
+
+@testset "Integrate for several variables" begin
+
+    t, x, y = set_variables("t x y")
+
+    @test integrate(t, 1) == 0.5*t^2
+    @test integrate(t, 2) == t * x
+    @test integrate(t, 3) == t * y
+    @test integrate(x, 1) == t * x
+    @test integrate(x, 2) == 0.5*x^2
+    @test integrate(y, 2) == x * y
+
 end

